cli/cmd/uninstall.go

```diff
@@ -0,0 +1,399 @@
+	if err != nil {
+		return err
+	}
+
+	resources, err := fetchKubernetesResources(k8sAPI)
+	if err != nil {
+		return err
+	}
+
+	for _, r := range resources {
+		if err := r.renderResource(os.Stdout); err != nil {
+			return fmt.Errorf("error rendering Kubernetes resource:%v", err)
+		}
+	}
+	return nil
+}
+
+func (r kubernetesResource) renderResource(w io.Writer) error {
+	b, err := yaml.Marshal(r)
+	if err != nil {
+		return err
+	}
+
+	_, err = w.Write(b)
+	if err != nil {
+		return err
+	}
+
+	_, err = w.Write([]byte(YamlSep))
+	return err
+}
+
+func fetchKubernetesResources(k *k8s.KubernetesAPI) ([]kubernetesResource, error) {
+	options := metav1.ListOptions{
+		LabelSelector: k8s.ControllerNSLabel,
+	}
+
+	resources := make([]kubernetesResource, 0)
+
+	clusterRoles, err := fetchClusterRoles(k, options)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch ClusterRole resources:%v", err)
+	}
+	resources = append(resources, clusterRoles...)
+
+	clusterRoleBindings, err := fetchClusterRoleBindings(k, options)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch ClusterRoleBinding resources:%v", err)
+	}
+	resources = append(resources, clusterRoleBindings...)
+
+	crds, err := fetchCustomResourceDefinitions(k, options)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch CustomResourceDefinition resources:%v", err)
+	}
+	resources = append(resources, crds...)
+
+	roleBindings, err := fetchKubeSystemRoleBindings(k, options)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch RoleBindings from kube-system namespace:%v", err)
+	}
+	resources = append(resources, roleBindings...)
+
+	apiCRDs, err := fetchAPIRegistrationResources(k, options)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch APIService CRDs:%v", err)
+	}
+	resources = append(resources, apiCRDs...)
+
+	psps, err := fetchPodSecurityPolicy(k, options)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch PodSecurityPolicy resources:%v", err)
+	}
+	resources = append(resources, psps...)
+
+	mutatinghooks, err := fetchMutatingWebhooksConfiguration(k, options)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch MutatingWebhookConfigurations:%v", err)
+	}
+	resources = append(resources, mutatinghooks...)
+
+	validationhooks, err := fetchValidatingWebhooksConfiguration(k, options)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch ValidatingWebhookConfiguration:%v", err)
+	}
+	resources = append(resources, validationhooks...)
+
+	namespace, err := fetchNamespaceResource(k)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch Namespace %s:%v", controlPlaneNamespace, err)
+	}
+
+	if namespace.Metadata.Name != "" {
+		resources = append(resources, namespace)
+	}
+
+	return resources, nil
+}
+
+func fetchClusterRoles(k *k8s.KubernetesAPI, options metav1.ListOptions) ([]kubernetesResource, error) {
+	list, err := k.RbacV1().ClusterRoles().List(options)
+	if err != nil {
+		return nil, err
+	}
+
+	if list.Items == nil {
```

Comment:
I believe that if a slice is nil, the for-loop will skip right over it, so I don't think we need this special case.